### PR TITLE
docs(website): fix typed rules heading link in custom rule docs

### PR DESCRIPTION
**Pull request by thilllon**
## PR Checklist

- [x] Addresses an existing open issue: fixes #000
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

The link is not matching. the link should address https://typescript-eslint.io/custom-rules#type-checking

### CODE REVIEW

File: docs/Custom_Rules.mdx [hunk 1]
```diff
@@ -14,7 +14,7 @@ The main three changes to custom rules writing are:
 
 - [Utils Package](#utils-package): we recommend using `@typescript-eslint/utils` to create custom rules
 - [AST Extensions](#ast-extensions): targeting TypeScript-specific syntax in your rule selectors
-- [Typed Rules](#typed-rules): using the TypeScript type checker to inform rule logic
+- [Typed Rules](#type-checking): using the TypeScript type checker to inform rule logic
```
Comment (JoshuaKGoldberg) [inline docs/Custom_Rules.mdx:17]: I think the right fix would be the other way around - that the section should be named _Typed Rules_ not _Type Checking_. Could you make that fix instead please?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Thanks for the typo spot! Just requesting changes to go with a different fix.

File: docs/Custom_Rules.mdx (hunk 1 above)
Comment (thilllon) [reply docs/Custom_Rules.mdx]: Thank you so much for your kind response... I fixed it as you mentioned.

**Review by JoshuaKGoldberg — APPROVED** [review]
✨ thanks!

**Pull request merged**